release/pkg/helm/helm.go

```diff
@@ -0,0 +1,409 @@
+// limitations under the License.
+
+package helm
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/eks-anywhere/release/pkg/clients"
+
+	"github.com/aws/eks-anywhere/release/pkg/aws/ecr"
+	"github.com/aws/eks-anywhere/release/pkg/aws/ecrpublic"
+
+	releasetypes "github.com/aws/eks-anywhere/release/pkg/types"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/registry"
+	"k8s.io/helm/pkg/chartutil"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	YamlSeparator = "\n---\n"
+)
+
+var BundleLog = ctrl.Log.WithName("BundleGenerator")
+
+// helmDriver implements PackageDriver to install packages from Helm charts.
+type helmDriver struct {
+	cfg      *action.Configuration
+	log      logr.Logger
+	settings *cli.EnvSettings
+}
+
+func CreateReleaseHelmDriver(r *releasetypes.ReleaseConfig) (*helmDriver, error) {
+	var sourceClients *clients.SourceClients
+	var releaseClients *clients.ReleaseClients
+	var err error
+	var authToken, authTokenPublic string
+
+	if r.ReleaseEnvironment == "development" {
+		sourceClients, releaseClients, err = clients.CreateStagingReleaseClients()
+		if err != nil {
+			return nil, errors.Wrap(err, "Error creating Staging Clients")
+		}
+		authToken, err = ecr.GetAuthToken(sourceClients.ECR.EcrClient)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error getting ecr Auth token Client")
+		}
+		authTokenPublic, err = ecrpublic.GetAuthToken(releaseClients.ECRPublic.Client)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error getting ecr public Auth token Client")
+		}
+	}
+
+	if r.ReleaseEnvironment == "production" {
+		sourceClients, releaseClients, err = clients.CreateProdReleaseClients()
+		if err != nil {
+			return nil, errors.Wrap(err, "Error creating Production Clients")
+		}
+		authToken, err = ecr.GetAuthToken(sourceClients.ECR.EcrClient)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error getting ecr Auth token Client")
+		}
+		authTokenPublic, err = ecrpublic.GetAuthToken(releaseClients.ECRPublic.Client)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error getting ecr public Auth token Client")
+		}
+	}
+	dockerReleaseStruct := &DockerAuth{
+		Auths: map[string]DockerAuthRegistry{
+			r.SourceContainerRegistry:  {Auth: authToken},
+			r.ReleaseContainerRegistry: {Auth: authTokenPublic},
+		},
+	}
+	dockerAuth, err := NewAuthFile(dockerReleaseStruct)
+	if err != nil || dockerAuth.Authfile == "" {
+		return nil, errors.Wrap(err, "Error creating Authfile")
+	}
+	driver, err := NewHelm(dockerAuth.Authfile)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error creating Helm Client")
+	}
+	return driver, nil
+}
+
+// NewAuthFile writes a new Docker Authfile from the DockerAuth struct which a user to be used by Skopeo or Helm.
+func NewAuthFile(dockerstruct *DockerAuth) (DockerAuthFile, error) {
```

Comment:
why do we need this when we have the AWS ECR credentials helper? I think Helm and Skopeo can use the default auth file at `$HOME/.docker/config.json`.

Reply:
I can test it out using that, we use this in our package controller tools since it's kinda wonky with helm.